### PR TITLE
luminous: Bring back diff -y for non-FreeBSD

### PR DESCRIPTION
**Pull request by smithfarm**
http://tracker.ceph.com/issues/24469
http://tracker.ceph.com/issues/24739

### CODE REVIEW

File: src/osd/PrimaryLogPG.cc [hunk 1]
```diff
@@ -348,21 +348,28 @@ void PrimaryLogPG::on_local_recover(
...
-    dout(20) << " snaps " << snaps << dendl;
-    snap_mapper.add_oid(
-      recovery_info.soid,
-      snaps,
-      &_t);
+      if (p != recovery_info.ss.clone_snaps.end()) {
+        snaps.insert(p->second.begin(), p->second.end());
+      } else {
+        derr << __func__ << " " << hoid << " had no clone_snaps" << dendl;
+        error = true;
```
Comment (dzafman) [inline src/osd/PrimaryLogPG.cc]: I don't thing you need the bool error.  Isn't there just an extra indentation, but make the same exact change as in master?

Comment (smithfarm) [reply src/osd/PrimaryLogPG.cc]: @dzafman No, IIRC you advised me to resolve the conflict this way.

Comment (smithfarm) [reply src/osd/PrimaryLogPG.cc]: because master doesn't have this `if (recovery_info.ss.is_legacy())`

Comment (dzafman) [reply src/osd/PrimaryLogPG.cc]: @smithfarm Yup, you/I was right.

**Comment by yuriw** [comment]
[wip-yuri2-testing-2018-08-01-1605-luminous](https://trello.com/c/r4uhYmse/361-wip-yuri2-testing-2018-08-01-1605-luminous)

**Pull request merged**